TickerQ.EntityFramework/DependencyInjection/ServiceExtension.cs

```diff
@@ -19,76 +22,83 @@ public static class ServiceExtension
+              
+                services.AddScoped<ITickerPersistenceProvider<TimeTicker, CronTicker>, TickerEFCorePersistenceProvider<TContext, TimeTicker, CronTicker>>();
             };
+
+            UseApplicationService(tickerConfiguration, efCoreOptionBuilder.CancelMissedTickersOnReset);
             
             return tickerConfiguration;
         }
 
-        /// <summary>
-        /// Timeout checker default is 1 minute, cannot set less than 30 seconds
-        /// </summary>
-        /// <param name="tickerConfiguration"></param>
-        /// <param name="timeSpan"></param>
-        public static TickerOptionsBuilder SetTimeOutJobChecker(this TickerOptionsBuilder tickerConfiguration, TimeSpan timeSpan)
+        private static DbContextOptions<TContext> UpdateDbContextOptionsService<TContext, TTimeTickerEntity, TCronTickerEntity>(IServiceProvider serviceProvider, Func<IServiceProvider, object> oldFactory) where TContext : DbContext where TTimeTickerEntity : TimeTickerEntity where TCronTickerEntity : CronTickerEntity
         {
-            tickerConfiguration.TimeOutChecker = timeSpan < TimeSpan.FromSeconds(30) 
-                ? TimeSpan.FromSeconds(30) 
-                : timeSpan;
+            var factory = (DbContextOptions<TContext>)oldFactory(serviceProvider);
 
-            return tickerConfiguration;
-        }
-        
-        
-        public static void CancelMissedTickersOnApplicationRestart(this TickerOptionsBuilder tickerConfiguration)
-        {
-            tickerConfiguration.CancelMissedTickersOnReset = true;
+            return new DbContextOptionsBuilder<TContext>(factory)
+                        .ReplaceService<IModelCustomizer, TickerModelCustomizer<TTimeTickerEntity, TCronTickerEntity>>()
+                        .Options;
         }
 
-
-        private static DbContextOptions<TContext> UpdateDbContextOptionsService<TContext, TTimeTicker, TCronTicker>(IServiceProvider serviceProvider, Func<IServiceProvider, object> oldFactory) where TContext : DbContext where TTimeTicker : TimeTicker where TCronTicker : CronTicker
+        private static void UseApplicationService(this TickerOptionsBuilder tickerConfiguration, bool cancelMissedTickersOnReset)
         {
-            var factory = (DbContextOptions<TContext>)oldFactory(serviceProvider);
+            tickerConfiguration.ExternalProviderConfigApplicationAction = (app) =>
+            {
+                using var scope = app.ApplicationServices.CreateScope();
 
-            return new DbContextOptionsBuilder<TContext>(factory)
-                        .ReplaceService<IModelCustomizer, TickerModelCustomizer<TTimeTicker, TCronTicker>>()
-                        .Options;
+                var internalTickerManager = scope.ServiceProvider.GetRequiredService<IInternalTickerManager>();
+                
+                var functionsToSeed = TickerFunctionProvider.TickerFunctions
+                    .Where(x => string.IsNullOrEmpty(x.Value.cronExpression) == false)
```

Comment:
@arcenox can't we just do `.Where(x => !string.IsNullOrEmpty(x.Value.cronExpression))` instead?